src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs

```diff
@@ -307,6 +309,10 @@ private void EmitInitializeMethod(ObjectSpec type)
                 {
                     if (property.ShouldBindTo && property.MatchingCtorParam is null)
                     {
+                        // Currently properties don't have an 'else' failure case. Doing that may collide with the
+                        // feature to set properties to their default values if they don't exist in the config section.
+                        Debug.Assert(property.ErrorOnFailedBinding == false);
```

Comment:
> // Currently properties don't have an 'else' failure case.

Seems like we should remove the abstraction to avoid these two features from clashing. Not sure that the property binding expectation here will change.

Reply:
I moved the ErrorOnFailedBindingProperty from the base class to the parameter class, and removed this assert.